File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int,int> map;
        for(int i=0;i<nums.size();i++){
            int numberToFind = target- nums[i];
            
            if(map.find(numberToFind)!= map.end()){
                return {map[numberToFind],i};
            }
            
            map[nums[i]] = i;
            
        }
        return {};
    }
};

File: 0001-two-sum/0001-two-sum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    const map = {};
    for(let i=0;i<nums.length;i++){
        const findNo = target - nums[i];
        if(map[findNo] != undefined){
            return [map[findNo],i];
        }
        map[nums[i]] = i;
    }
    return [-1,-1]
};